test/test_custom_ops.py

```diff
@@ -1534,6 +1534,179 @@ def foo_impl(x):
+                    return op(x)
+
+            @staticmethod
+            def backward(ctx, grad):
+                raise NotImplementedError()
+
+        def autograd_impl(x):
+            return Op.apply(x)
+
+        lib.impl(name, autograd_impl, "Autograd")
+        return op
+
+    def _op_with_no_abstract_impl(self, name):
+        lib = self.lib()
+        lib.define(f"{name}(Tensor x) -> Tensor")
+        qualname = f"{self.test_ns}::{name}"
+        lib.impl(name, lambda x: x.clone(), "CPU")
+        return self.get_op(qualname)
+
+    def test_mm(self):
+        x = torch.randn(2, 3, requires_grad=True)
+        y = torch.randn(3, 5)
+        result = torch.ops.aten.mm.default(x, y)
+        self.assertEqual(result, x @ y)
+
+    def test_mm_errors(self):
+        x = torch.randn(2, 3, requires_grad=True)
+        y = torch.randn(4, 5)
+        with self.assertRaisesRegex(RuntimeError, "cannot be multiplied"):
+            result = torch.ops.aten.mm.default(x, y)
+
+    def test_nonzero(self):
+        x = torch.tensor([0, 1, 2, 0, 0])
+        y = torch.ops.aten.nonzero.default(x)
+        self.assertEqual(y, torch.tensor([[1], [2]]))
+
+    def test_inplace(self):
+        x = torch.randn(3)
+        x_clone = x.clone()
+        y = torch.ops.aten.sin_(x)
+        self.assertEqual(x, x_clone.sin())
+
+    def test_incorrect_schema(self):
+        op = self._op_with_incorrect_schema("incorrect_schema")
+        x = torch.randn(3)
+        op(x)
+
+    def test_no_abstract(self):
+        op = self._op_with_no_abstract_impl("no_abstract")
+        x = torch.randn(3)
+        op(x)
+
+    def test_delayed_error(self):
+        op = self._op_delayed_backward_error("delayed_error")
+        x = torch.randn([], requires_grad=True)
+        y = op(x)
+        with self.assertRaises(NotImplementedError):
+            y.sum().backward()
+
+    def test_delayed_error_no_requires_grad(self):
+        op = self._op_delayed_backward_error("delayed_error")
+        x = torch.randn([])
+        y = op(x)
+
+
+mini_op_test_failures_dict = {
+    "aten::nonzero": {
+        # Nonzero doesn't support static shapes
+        "test_aot_dispatch_static__test_nonzero": "xfail",
+    },
+    "mini_op_test::delayed_error": {
+        "test_aot_dispatch_dynamic__test_delayed_error": "xfail",
+        "test_aot_dispatch_static__test_delayed_error": "xfail",
+    },
+    "mini_op_test::incorrect_schema": {
+        # "skip" just to test the skip mechanism
+        "test_schema__test_incorrect_schema": "skip",
+    },
+    "mini_op_test::no_abstract": {
+        "test_aot_dispatch_dynamic__test_no_abstract": "xfail",
+        "test_aot_dispatch_static__test_no_abstract": "xfail",
+        "test_faketensor__test_no_abstract": "xfail",
+    },
+}
+
+mini_op_test_checks = [
+    "test_schema",
+    "test_autograd_registration",
+    "test_faketensor",
+    "test_aot_dispatch_static",
+    "test_aot_dispatch_dynamic",
+]
+
+optests.generate_opcheck_tests(
```

Comment:
Help wanted: There are two testing designs we can do.
1. This PR: we activate cross-ref testing on a TestCase.
2. We manually run the TestCase to grab sample inputs and generate a Python file that has the sample inputs in it. Then, we set up some OpInfo-style testing that ingests the generated sample inputs. The Python file that has the sample inputs will get checked into the repository (instead of being autogenerated).

This PR goes the first approach, but I really wish we could make the second approach actually work